frontend/src/metabase/writeback/containers/ActionParametersInputForm/utils.ts

```diff
@@ -53,6 +54,15 @@ export const getInitialValues = (
   prefetchValues: ParametersForActionExecution,
 ) => {
   return Object.fromEntries(
-    form.fields.map(field => [field.name, prefetchValues[field.name] ?? ""]),
+    form.fields.map(field => {
+      const value = prefetchValues[field.name];
+      const formattedValue =
+        field.type === "date"
+          ? moment(value).utc(false).format("YYYY-MM-DD")
+          : field.type === "datetime-local"
+          ? moment(value).utc(false).format("YYYY-MM-DDTHH:mm")
+          : value ?? "";
```

Comment:
This is a bit hard to parse IMO, can we do something about it to make it a easier to follow? Maybe extract it to a function like `formatValue` or something?